### DISCUSSION TITLE
Add options to simplify() to exclude specific simplifications

### DISCUSSION THREAD

**Discussion by niklasrb** [sympy/sympy#25961] [Ideas]
Hey everyone,
I've been playing around with einsteinpy, and I wanted to calculate the Ricci Tensor for the Kerr metric (modifying [this example](https://github.com/einsteinpy/einsteinpy/blob/main/docs/source/examples/Ricci%20Tensor%20and%20Scalar%20Curvature%20symbolic%20calculation.ipynb) ). The computation gets stuck on a .simplify() during the trigsimp call and didn't finish after a couple days of computation. I guess the expressions do get pretty complicated. 
I've modified my personal version as to have a simplify(..., usetrigsimp=True) function, where I can optionally turn off the trigsimp call, which fixed the problem.
In the documentation you say that when knowing about the expression a specific simplify call (nsimplify, trigsimp, etc.) is usually suited better, and I agree with that. But for general expressions, it might be useful to turn off just one simple specific simplify call to leave everything else intact. So my suggestion is to include kwargs such as usetrigsimp, usebesselsimp, usecombsimp in the simplify function.

Thanks and best,
Niklas

**Comment by oscarbenjamin** [comment]
Can you show the code for which simplify is slow?

**Reply by niklasrb** [reply]
```
import sympy
from einsteinpy.symbolic import ChristoffelSymbols, RiemannCurvatureTensor
from einsteinpy.symbolic.predefined import Kerr

metric = Kerr(c=1)
chris = ChristoffelSymbols.from_metric(metric)
riemannTensor = RiemannCurvatureTensor.from_christoffels(chris)
```

For me, the execution gets stuck [here](https://github.com/einsteinpy/einsteinpy/blob/main/src/einsteinpy/symbolic/riemann.py#L90) during a simplify call of the trigsimp subroutine and doesn't return after a day of computation.

**Reply by oscarbenjamin** [reply]
I don't think that this should call simplify unconditionally. Based on the given example the expressions are polynomial trig expressions so probably some combination of fu functions and something like cancel or factor would be best here rather than simplify.

The thing that is slow seems to be `dmp_zz_wang` or at least it is being called many times. Sometimes `dmp_zz_wang` can just be randomly slow because of poor seed choice (gh-20456). Unfortunately flint cannot yet be used for multivariate factorisation (we need the sparse polynomials).
